### PR TITLE
HTS-1229 Adds manual override to ineligible screen

### PR DESCRIPTION
**Pull request by thebailers**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala [hunk 1]
```diff
@@ -181,17 +187,66 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
                 Ok(views.html.account_already_exists())
             })
         }
+      },
+                 whenIneligible = { (ineligible, nSIUserInfo) ⇒
+        ineligible.manualCreationAllowed match {
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: its an overkill to use pattern matching for booleans

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala [hunk 2]
```diff
@@ -59,7 +59,12 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
 
   private def checkIfAlreadyEnrolled(nino: String)(ifNotEnrolled: ⇒ Future[Result])(implicit request: Request[AnyContent], hc: HeaderCarrier): Future[Result] = { // scalastyle:ignore
       def updateSessionIfEnrolled(enrolmentStatus: EnrolmentStatus)(implicit hc: HeaderCarrier): EitherT[Future, String, Unit] = enrolmentStatus match {
-        case Enrolled    ⇒ keyStoreConnector.put(HtsSession(AlreadyHasAccount)).map(_ ⇒ ())
+        case Enrolled ⇒ {
+          helpToSaveConnector.getNSIUserInfo(nino).map { nsiUserInfo ⇒
+            val x = keyStoreConnector.put(HtsSession(AlreadyHasAccount, nsiUserInfo))
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: can we give this a better name  ans handle response of  `put ` ?

Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: @Jackie1988 ^^^

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala (hunk 1 above)
Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: @Jackie1988 ^^^

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/SessionBehaviour.scala [hunk 3]
```diff
@@ -50,66 +50,69 @@ trait SessionBehaviour {
       ).flatMap(identity)
 
   def checkSession(noSessionData:         ⇒ Future[Result],
-                   whenEligible:          (EligibleWithNSIUserInfo, Boolean) ⇒ Future[Result] = (_, _) ⇒ SeeOther(routes.StrideController.customerEligible().url),
-                   whenIneligible:        Ineligible ⇒ Future[Result] = _ ⇒ SeeOther(routes.StrideController.customerNotEligible().url),
-                   whenAlreadyHasAccount: () ⇒ Future[Result]                                 = () ⇒ SeeOther(routes.StrideController.accountAlreadyExists().url)
+                   whenEligible:          (EligibleWithNSIUserInfo, Boolean, NSIUserInfo) ⇒ Future[Result] = (_, _, _) ⇒ SeeOther(routes.StrideController.customerEligible().url),
+                   whenIneligible:        (Ineligible, NSIUserInfo) ⇒ Future[Result]                       = (_, _) ⇒ SeeOther(routes.StrideController.customerNotEligible().url),
+                   whenAlreadyHasAccount: (NSIUserInfo) ⇒ Future[Result] = _ ⇒ SeeOther(routes.StrideController.accountAlreadyExists().url)
```
Comment (AndyHWChung) [inline app/uk.gov.hmrc.helptosavestridefrontend/controllers/SessionBehaviour.scala]: the brackets around `NSIUserInfo` are not needed here

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: ok sure

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala (hunk 2 above)
Comment (Jackie1988) [reply app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: updated, thanks

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala (hunk 1 above)
Comment (Jackie1988) [reply app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: updated, thanks

**Pull request merged**